.travis.yml

```diff
@@ -24,6 +24,7 @@ install:
 script:
   - luacheck .
   - lua run.lua tests --luacov
+  - lua run.lua examples
```

Comment:
I think we should have a single line here;

    lua run.lua --luacov

this will run everything, and add coverage reporting for everything as well.

Reply:
I deliberately ran it in two steps because then it is clean from the Travis output log what failed. The output isn't very clear when they are both run together.

I did wonder about adding coverage to the examples, but I tend to thing that it would be better to measure test coverage via the smallest unit testing possible, big examples that cover lots of ground can catch some issues but they don't do a great job at identifying specific expectations of each interface along the way. I would rather specific unit tests be the baseline for coverage reporting.